app/models/notification.rb

```diff
@@ -0,0 +1,33 @@
+class Notification < ActiveRecord::Base
+  include Versions # version/notification_version.rb
+
+  belongs_to :registrar
+  belongs_to :action
+
+  scope :unread, -> { where(read: false) }
+
+  validates :text, presence: true
+
+  after_initialize :set_defaults
+
+  def mark_as_read
+    raise 'Read notification cannot be marked as read again' if read?
```

Comment:
This is quite a significant change in behavior from `dequeue `. Shouldn't this be a no-op instead?

Reply:
What does "no-op" stand for? How an exception is a significant change (if you mean this)? Can't find anything specific that case in EPP spec.

Reply:
Probably because [no-op](https://en.wikipedia.org/wiki/NOP) has nothing to do with EPP. 

Exception is a significant change because previous implementation did not throw an error when dequeuing at any point. https://github.com/internetee/registry/blob/master/app/models/message.rb#L11-L14

Reply:
What I mean is that:
- EPP poll is covered with tests.
- Making a read notification as read repeatedly is neither documented nor supported.
- Timo will additionally verify that it works as expected. I see no problem making additional fixing, if it is not.